src/Microsoft.AspNetCore.OData/Query/Query/DefaultSkipTokenHandler.cs

```diff
@@ -136,7 +136,12 @@ internal static string GenerateSkipTokenValue(Object lastMember, IEdmModel model
                 string clrPropertyName = model.GetClrPropertyName(edmProperty);
                 if (obj != null)
                 {
-                    obj.TryGetPropertyValue(clrPropertyName, out value);
+                    if (!obj.TryGetPropertyValue(
+                            clrPropertyName,
+                            out value))
+                    {
+                        obj.TryGetPropertyValue(edmProperty.Name, out value);
+                    }
```

Comment:
Thank you @orty for your pull request contribution.
I'd propose a simpler fix to this issue:
```suggestion
                    obj.TryGetPropertyValue(edmProperty.Name, out value);
```
I did a quick test and confirmed that it does the job.
If `lastMember` can be cast to `IEdmStructuredObject`, it should be okay to use the Edm property name to retrieve the property value from the object.
I didn't go to the extent of running all tests to confirm that nothing broke - let's try the change and see how it goes.

Note: We can also move the declaration for `clrPropertyName` to the `else` block since that's the only place where it'd be applicable/relevant.

Reply:
I did the change you suggested and I confirm it works like a charm. I went ahead and added some tests to check if an aliased model could also be handled like that, and discovered something.

In here :
https://github.com/OData/AspNetCoreOData/blob/118c2c175f49ebcf47fab64840bf4a8a675e8e8f/src/Microsoft.AspNetCore.OData/Query/Query/DefaultSkipTokenHandler.cs#L431-L435

we should retrieve the proper underlying CLR type property name for the `IEdmProperty`, like it is done in the `QueryBinder.CreatePropertyAccessExpression` method, by adding `context.Model.GetClrPropertyName(property);`.

This is because, when the value-pairs dictionary has been populated, today it only works by chance that the `Expression` can be built here, in the `ApplyToCore` method
https://github.com/OData/AspNetCoreOData/blob/118c2c175f49ebcf47fab64840bf4a8a675e8e8f/src/Microsoft.AspNetCore.OData/Query/Query/DefaultSkipTokenHandler.cs#L291-L296

It works with both PascalCased and LowerCamelCased propery names because of the permissivity of LINQ. When an alias is provided, it just throws an exception. With the next commit I will push this afternoon (it is 1PM here, gotta take a break ;)), the `Expression` is able to be built even with an alias.

I added all the subsequent unit tests for this change.